src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/AmiNameResolver.kt

```diff
@@ -26,24 +43,37 @@ internal class AmiNameResolver {
                 .images
                 .map { it.imageId }
                 .singleOrNull()
-                ?: throw Exception("Failed to find image $vuAmiName in ${awsAndAmi.aws.region}")
+                ?: throw Exception("Failed to find image $vuAmiName in ${awsAndAmiName.aws.region}")
         }
-    }
 
-    data class AwsAndAmi constructor(val aws: Aws, val ami: String) {
+        private fun getVuAmiName(architecture: String): String {
+            val imgArchitecture = if (architecture != "arm64") {
+                "amd64"
+            } else {
+                architecture
+            }
+
+            return "ubuntu-focal-20.04-${imgArchitecture}-server-20200701"
```

Comment:
Hmmm. I've just realised that uses of aws-infrastructure will be broken if they use SDK older than the one where they introduced DescribeInstanceTypes. How do we handle that? That means a new major release, right?

Reply:
That's the problem with hiding version conflicts by default (damn you Maven). We can express that aws-infra requires at least this version, but it could be ignored.
Gradle consumers are in the clear. They can `failOnVersionConflict` to detect it early. And they can use dependency locks to control when the bump (and conflict) happens.
I tried to depend on AWS SDK SemVer range, but Maven would download every single POM for every version in the range (hundreds). It increased Maven build times by tens of minutes on cold caches. Gradle did no such thing.
Our examples for vendors use Maven, because the Atlassian ecosystem is built upon it. 

So in general, it's not a problem for Gradlers, and it might be a problem for Maveners.
Even the loose Gradle users (without dep locks and strict conflicts) would not be impacted, because Gradle would pick the higher version anyway. Maven will pick the closest dep in the tree.

By accident, `jira-performance-tests` depends on `aws-infrastructure` first and then on `aws-resources`, so aws-infra AWS SDK will win. If we ever need to take the control of this, `jira-performance-test` can resolve this conflict in its POM.

So for now, we can stick to the [current POM contracts](https://github.com/atlassian/aws-infrastructure/blob/37f70fc32fa801cbe7df73a2921b5d79e80544c0/CHANGELOG.md#pom):
> Adding a requirement of a major version of a dependency is breaking a contract. Dropping a requirement of a major version of a dependency is a new contract.

If we ever find a way to (any of the following):
* get `failOnVersionConflict` for Maven
* get dependency locks for Maven (this is actually very probable)
* get Maven to stop downloading the entire dependency version range
* avoid Maven as a build tool

... then we can rest easy with the current POM contracts.
A better POM contract could help us too, but I'm out of ideas there.